src/diagnostics/diagnostic-structs.md

```diff
@@ -116,17 +116,24 @@ In the end, the `SessionDiagnostic` derive will generate an implementation of
 impl SessionDiagnostic for FieldAlreadyDeclared {
     fn into_diagnostic(self, sess: &'_ rustc_session::Session) -> DiagnosticBuilder<'_> {
         let mut diag = sess.struct_err_with_code(
-            rustc_errors::DiagnosticMessage::fluent("typeck-field-already-declared"),
+            rustc_errors::DiagnosticMessage::Identifier(
```

Comment:
Sorry, I meant that this should be `rustc_errors::fluent::whatever::the_user_wrote`, the macros don't generate code that uses strings to refer to slugs at all now.

Reply:
Is that what the current implementation is doing? Because currently it's doing [this](https://github.com/rust-lang/rust/blob/4916e2b9e6ef8cee6f9c6abb75bd01ba9dc07e5c/compiler/rustc_macros/src/diagnostics/fluent.rs#L201-L207).

Reply:
(And yes, it should be `FluentIdentifier`, I have to adjust that)

Reply:
Yeah, that's what the `fluent_messages` macro is doing, but the diagnostic derive macros generate the typed identifiers, see https://github.com/rust-lang/rust/blob/8556e6620e4866526b3cea767ad8c20ae877a569/compiler/rustc_macros/src/diagnostics/diagnostic.rs#L69